### ISSUE TITLE
possible false positive with RET504 and class method

### ISSUE THREAD

**Issue by Borda** [astral-sh/ruff#4173]
I have the following code sample, and here I do think it is correct:
```py
  def collate_none_when_even(self, batch):
      result = None if self.counter % 2 == 0 else default_collate(batch)
      self.counter += 1
      return result
```
or another example:
```py
def training_step(self, dataloader_iter: Iterator) -> STEP_OUTPUT:
    output = super().training_step(dataloader_iter)
    batch_idx = self.trainer.fit_loop.epoch_loop.batch_idx
    if self.trigger_stop_iteration and batch_idx == EXPECT_NUM_BATCHES_PROCESSED:
        raise StopIteration
    return output
```

**Comment by charliermarsh** [comment]
I'm considering changing this rule to only detect the simple case of "assignment followed by return".
